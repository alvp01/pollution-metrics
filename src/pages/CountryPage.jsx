import { Link } from 'react-router-dom';
import Country from '../components/Country';

const CountryPage = () => (
  <div className="container-fluid">
    <Link to="/">Back</Link>
    <Country />
  </div>
);

export default CountryPage;
