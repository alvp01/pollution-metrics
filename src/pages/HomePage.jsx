import CountriesList from '../components/CountriesList';

const HomePage = () => (
  <>
    <CountriesList />
  </>
);

export default HomePage;
