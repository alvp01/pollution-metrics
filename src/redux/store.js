import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
  middleware: [thunk, logger],
});

export default store;
