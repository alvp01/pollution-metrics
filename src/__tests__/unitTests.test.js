import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
  Routes,
  Route,
} from 'react-router-dom';
import HomePage from '../pages/HomePage';
import CountryPage from '../pages/CountryPage';

import store from '../redux/store';

afterEach(cleanup);

describe('test components', () => {
  it('renders homepage page with redux', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route element={<HomePage />}/>
          </Routes>
        </Router>        
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders country page with redux', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route element={<CountryPage />}/>
          </Routes>
        </Router>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
