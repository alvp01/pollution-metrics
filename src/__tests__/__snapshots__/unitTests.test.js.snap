// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components renders country page with redux 1`] = `<DocumentFragment />`;

exports[`test components renders homepage page with redux 1`] = `<DocumentFragment />`;
